File: SoobinJung/Lv2/올바른 괄호.java
import java.util.*;

class Solution {

  boolean solution(String s) {
    Stack<Character> st = new Stack<>();

    for (int i = 0; i < s.length(); i++) {
      if (s.charAt(i) == '(') {
        st.push(s.charAt(i));
      } else {
        if (st.isEmpty()) {
          return false;
        }
        st.pop();
      }
    }
    return st.isEmpty();
  }
}

File: 정인희/LV2/올바른 괄호.py
def solution(s):
    answer = True
    s=list(s)
    if len(s)%2!=0 or s[0]!='(':
        return False
    stack=[]
    for i in s:
        if i=='(':
            stack.append(i)
        else:
            if stack:
                stack.pop()
            else:
                return False
    if len(stack)==0:
        return True
    else:
        return False
